components/remote-environment-broker/internal/broker/middleware_test.go

```diff
@@ -0,0 +1,88 @@
+package broker
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/kyma-project/kyma/components/remote-environment-broker/internal/broker/automock"
+	"github.com/kyma-project/kyma/components/remote-environment-broker/platform/logger/spy"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestOSBContextForClusterScopedBroker(t *testing.T) {
+	// GIVEN
+	mockBrokerFlavor := &automock.BrokerFlavorProvider{}
+	defer mockBrokerFlavor.AssertExpectations(t)
+
+	mockBrokerFlavor.On("IsClusterScoped").Return(true)
+	sut := NewOsbContextMiddleware(mockBrokerFlavor, spy.NewLogDummy())
+
+	req := httptest.NewRequest(http.MethodGet, "https://core-reb.kyma-system.svc.cluster.local/v2/catalog", nil)
+	rw := httptest.NewRecorder()
+	nextCalled := false
+	// WHEN
+	sut.ServeHTTP(rw, req, func(nextRw http.ResponseWriter, nextReq *http.Request) {
+		nextCalled = true
+		osbCtx, ex := osbContextFromContext(nextReq.Context())
+		// THEN
+		assert.True(t, ex)
+		assert.True(t, osbCtx.ClusterScopedBroker)
```

Comment:
should we also add:
```go
assert.Empty(t, osbCtx.BrokerNamespace)
```
is there a contract which says if ClusterScopedBroker is set to `true` then BrokerNamespace should not be populated? IMO should